src/mono/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.Mono.cs

```diff
@@ -44,6 +44,12 @@ public static int GetHashCode(object? o)
             return InternalGetHashCode(o);
         }
 
+        internal static int TryGetHashCode(object o)
+        {
+            // On Mono the fast path is not yet implemented, so we just defer to GetHashCode.
```

Comment:
I do not see anything in the Mono hashcode implementation that guarantees the assigned hashcode to be non-zero.

Reply:
Oh dear. I'll if-def it out for Mono.